File: backend/utils/validators.py
def validate_code(code: str) -> bool:
    return bool(code and len(code.strip()) > 0)


def validate_language(language: str) -> bool:
    supported = ["python", "javascript", "typescript", "java", "go", "cpp", "rust"]
    return language.lower() in supported


def validate_filename(filename: str) -> bool:
    return bool(filename and "." in filename)

File: frontend/src/utils/validators.js
export function isValidCode(code) {
  return code && code.trim().length > 0;
}

export function isValidLanguage(language) {
  const supported = ['python', 'javascript', 'typescript', 'java', 'go', 'cpp'];
  return supported.includes(language.toLowerCase());
}

export function isValidFileSize(file, maxSize = 10 * 1024 * 1024) {
  return file.size <= maxSize;
}
